src/main/java/com/flightstats/hub/webhook/LocalWebhookRunner.java

```diff
@@ -87,31 +88,58 @@ private boolean ensureRunningWithLock(String name) {
 
     private boolean start(Webhook daoWebhook) {
         WebhookLeader webhookLeader = webhookLeaderProvider.get();
-        boolean hasLeadership = webhookLeader.tryLeadership(daoWebhook);
-        if (hasLeadership) {
-            localLeaders.put(daoWebhook.getName(), webhookLeader);
-        }
-        return hasLeadership;
+        return webhookLeader.tryLeadership(daoWebhook, localLeaders);
     }
 
     void stopAll() {
-        runAndWait("LocalWebhookRunner.stopAll", localLeaders.keySet(), this::stop);
+        runAndWait("LocalWebhookRunner.stopAll", localLeaders.getLocalLeaders().keySet(), this::stop);
     }
 
     void stop(String name) {
+        lockManager.executeLocked(name, () -> stopWithLock(name));
+    }
+
+    private void stopWithLock(String name) {
         log.info("stopping {} if running local", name);
-        if (localLeaders.containsKey(name)) {
+        WebhookLeader webhookLeader = localLeaders.getLocalLeaders().get(name);
+        if (null != webhookLeader) {
             log.info("stopping local {}", name);
-            localLeaders.get(name).exit();
-            localLeaders.remove(name);
+            webhookLeader.exit();
         }
     }
 
     int getCount() {
-        return localLeaders.size();
+        return localLeaders.getLocalLeaders().size();
+    }
+
+    Map<String, Map<String, Object>> getRunning() {
+        return localLeaders.getLocalLeaders().values().stream()
+                .collect(Collectors.toMap(k -> k.getWebhook().getName(), WebhookLeader::getLocalStatistics));
     }
 
-    List<String> getRunning() {
-        return new ArrayList<>(localLeaders.keySet());
+
+    static class LocalLeaders implements WebhookLeader.LeadershipStateListener {
+        private final Map<String, WebhookLeader> localLeaders = new ConcurrentHashMap<>();
+
+        public Map<String, WebhookLeader> getLocalLeaders() {
+            return localLeaders;
+        }
+
+        @Override
+        public void leadershipStateUpdated(WebhookLeader webhookLeader, boolean hasLeadership) {
+            String webhookName = webhookLeader.getWebhook().getName();
+
+            if (hasLeadership) {
+                WebhookLeader existingLeader = localLeaders.put(webhookName, webhookLeader);
+                if (null != existingLeader) {
+                    log.error("Starting a second webhook leader for {}", webhookName);
```

Comment:
If we can get access to the datadog/statsd client in here, would it make sense to record two new metrics so that we can monitor them outside of the endless oceans that are our hub logs?